### PR TITLE
Update erc-7007.md to add opML compatibility

### PR DESCRIPTION
**Pull request by tina1998612**
When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: ERCS/erc-7007.md [hunk 1]
```diff
@@ -133,6 +139,10 @@ ERC-7007 Metadata JSON Schema for reference
         "aigc_data": {
             "type": "string",
             "description": "A URI pointing to a resource with mime type image/* representing the asset to which this AIGC NFT represents."
+        },
+        "verificaton": {
+            "type": "string",
+            "description": "Specify the verification method. Input zkML or opML."
```
Comment (xhyumiracle) [inline ERCS/erc-7007.md:145]: I think this makes sense. want to propose another possible wording:
"proof_type": {
  "type": "validity" / "fraud",
  ...
}
the point here is to align with "aigc_type".
we can discuss this.

File: ERCS/erc-7007.md [hunk 2]
```diff
@@ -47,33 +52,34 @@ interface IERC7007 is IERC165, IERC721 {
...
 
     /**
-     * @dev Verify the `prompt`, `aigcData` and `proof`.
+     * @dev Verify the `prompt`, `aigcData` and `proofOrState`.
      */
     function verify(
         bytes calldata prompt,
         bytes calldata aigcData,
-        bytes calldata proof
+        bytes calldata proofOrState
```
Comment (xhyumiracle) [inline ERCS/erc-7007.md]: this is a good way to integrate. just a question:
I assume the 'state' means the state root hash in opml, provided by the submitter, but then can we still call it 'verify'? would like to see how does the opml challenger interact with this interface.

File: assets/erc-7007/contracts/ERC7007_opml.sol [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                uri,
+                ', "prompt": "',
+                string(prompt),
+                '", "aigc_data": "',
+                string(aigcData),
+                '"}'
+            )
+        );
+        _setTokenURI(tokenId, tokenUri);
+        tokenIdToRequestId[tokenId] = IOpmlLib(opmlLib).initOpmlRequest(keccak256(prompt));
```
Comment (0x1cc) [inline assets/erc-7007/contracts/ERC7007_opml.sol]:
```
IOpmlLib(opmlLib).uploadResult(tokenIdToRequestId[tokenId], keccak256(aigcData));
```

File: assets/erc-7007/contracts/ERC7007_opml.sol [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    /**
+     * @dev See {IERC7007-verify}.
+     */
+    function verify(
+        bytes calldata prompt,
+        bytes calldata aigcData,
+        bytes calldata proof
+    ) public view virtual override returns (bool success) {
+        uint256 tokenId = uint256(keccak256(prompt));
+        return IOpmlLib(opmlLib).isFinalized(tokenIdToRequestId[tokenId]);
```
Comment (0x1cc) [inline assets/erc-7007/contracts/ERC7007_opml.sol]:
```
uint256 tokenId = uint256(keccak256(prompt));
bytes32 output = IOpmlLib(opmlLib).getOutputHash(tokenIdToRequestId[tokenId]);
return IOpmlLib(opmlLib).isFinalized(tokenIdToRequestId[tokenId]) && (output == keccak256(aigcData));
```

**Review by 0x1cc — COMMENTED** [review]
we should utilize the output (aigcData) in function mint and verify to check the result.

File: assets/erc-7007/contracts/IOpmlLib.sol [hunk 5]
```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.18;
+
+interface IOpmlLib {
+    	function initOpmlRequest(bytes32 inputHash) external returns (uint256 requestId); // we can construct the initialState using the input, can replace input with prompt (string)
```
Comment (0x1cc) [inline assets/erc-7007/contracts/IOpmlLib.sol]:
```
function initOpmlRequest(bytes calldata input) external returns (uint256 requestId);
```

File: assets/erc-7007/contracts/IOpmlLib.sol [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+	function uploadResult(uint256 requestId, bytes32 output) external;
```
Comment (0x1cc) [inline assets/erc-7007/contracts/IOpmlLib.sol]:
```
function uploadResult(uint256 requestId, bytes calldata output) external;
```

File: assets/erc-7007/contracts/IOpmlLib.sol [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+	function startChallenge(uint256 requestId, bytes32 finalState) external returns (uint256 challengeId);
+	function respondState(uint256 challengeId, bytes32 stateHash) external;
+	function proposeState(uint256 challengeId, bytes32 stateHash) external;
+	function assertStateTransition(uint256 challengeId) external;
+    	function isFinalized(uint256 requestId) external view returns (bool);
+	function getOutputHash(uint256 requestId) external view returns (bytes32 output);
```
Comment (0x1cc) [inline assets/erc-7007/contracts/IOpmlLib.sol]:
```
function getOutput(uint256 requestId) external view returns (bytes memory output);
```

**Review by 0x1cc — COMMENTED** [review]
replace bytes32 (input/output) in `IOpmlLib.sol` with bytes (we cann allow upload the whole data or only the hash)

**Pull request merged**